osbuild/util/ostree.py

```diff
@@ -215,16 +215,30 @@ def parse_input_commits(commits):
     return commits["path"], data["refs"]
 
 
-def deployment_path(root: PathLike, osname: str = "", ref: str = "", serial: int = None):
-    """Return the path to a deployment given the parameters"""
+def parse_deployment_option(root, deployment):
+    """Parse the deployment option and return the osname, ref, and serial"""
 
-    if osname == "" and ref == "" and serial is None:
+    default_deployment = deployment.get("default", False)
+    if default_deployment:
         filenames = glob.glob(os.path.join(root, 'ostree/deploy/*/deploy/*.0'), recursive=True)
         if len(filenames) < 1:
             raise ValueError("Could not find deployment")
         if len(filenames) > 1:
             raise ValueError("More than one deployment found")
-        return filenames[0]
+        
+        path = filenames[0].split("/")
+        osname = path[-3]
+        ref, serial = path[-1].split(".")
+        return osname, ref, serial
```

Comment:
note that the `ref` we are returning here is the commit ID of the deployment. I don't know if that will have a material impact or not, but it could..

The ref we have been using throughout before this is `ostree/0/1/0`. I think we have 3 options here:

1. do some investigation and decide using the commit is OK
2. hardcode `ostree/0/1/0` here (not sure of the ramifications of that)
3. do some sort of reverse lookup (opposite of `ostree rev-parse`) to see what ref points to the commit we found

For `3.` this works to just print out the commits each ref points to but I don't know if we really need to do something like this or not.

```
$ ostree refs | xargs -I{} bash -c "echo -n '{} '; ostree rev-parse {}"
fedora-compose:fedora/aarch64/coreos/next 639b5b1a12b49977d98de014c82e3017f0ceb6c054b0dc53e82cdfd6c587f95e
fedora:fedora/aarch64/coreos/next 33d5d167e6edd1ec7a3060f5208e868356609306dba0260841049996327e8fa7
ostree/0/1/0 7c923977bcc5f1a58fb0b174fb1d5756d9dfa83d311ff95ed61bfa152765d9fe
ostree/0/1/1 aa621168d0a62f56ed8591e35c6cebd29bd57da7f77f3dd720477eb586c0c4ac
rpmostree/base/0 33d5d167e6edd1ec7a3060f5208e868356609306dba0260841049996327e8fa7
rpmostree/base/1 d5b78e63c9f20056b2ab3bfa3f99d8b66d3bb087e3fce6630999c7c05feac87c
rpmostree/base/2 23ab01229a932eaefa10288dca1448cfbaf368138b163ee54e0fcac045a0001b
rpmostree/pkg/tailscale/1.54.1-1.aarch64 bb7e0e5896e6d140adb4ba3b7216691f6932ae07cfd3ce52e2435e7c4a929aa4
rpmostree/pkg/tailscale/1.56.1-1.aarch64 516015ce9fc2942a0c9b2ad49589228b9da0075b3158e279df7aa6aba8c85092
```

Reply:
also here we may want to convert the serial into an `int` when we return it.

Reply:
AFAICT, all the callers of this function use `ref` to just eventually build the deployment path. Which seems odd since this function has that path already right there, so it could also return that so it doesn't need to be rebuilt.

If you want to make the function generic, the best ref to use would be the actual `refspec` setting in the origin file sitting alongside the deployment dir that was globbed. And I think IIRC there's already code somewhere that parses the origin file.

Reply:
I think `refspec` (the literal refspec) is only in the origin file if we didn't deploy via container

Reply:
Maybe simplest is to rename that variable `commit` in all the callers so it's clear that it's the commit, not the ref. If any code needs the actual ref, then it can decide what to do in the container ref vs ostree ref cases.

Reply:
But honestly, are there cases today where we expect multiple deployments to exist? ISTM like to start we should just drop those osname/commit/serial knobs and always auto-detect the deployment path.

Reply:
yeah, I agree, but OSBuild tries to be super backwards compatible so I doubt they'll go for dropping those arguments.

Reply:
@lukewarmtemp I say let's just return the commit (like you are doing here) and we'll find any problems with it and fix it later.

Reply:
I'd still call the return value `commit` at least to be clear.

Reply:
SGTM